flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -210,19 +212,35 @@ private boolean isStreamingSource() {
 								" cannot be less than 1");
 			}
 
-			int splitNum;
 			try {
-				long nano1 = System.nanoTime();
-				splitNum = inputFormat.createInputSplits(0).length;
-				long nano2 = System.nanoTime();
+				// `createInputSplits` is costly,
```

Comment:
Can you extract these parallelism inference logic to a dedicated class like `HiveParallelismInference` (Including limit logic)?

Reply:
I don't think this is needed, at least in this PR. This PR is only an optimization and code cleanups should be done in separate PRs.

Reply:
Yes, it can be a separate PR.
You are making the logic more complex, so I think you should cleanup code first.